### PR TITLE
Enable staticcheck and fix all issues

### PR DESCRIPTION
**Pull request by Jacalz**
This PR makes sure to get all the issues reported by staticcheck fixed and marks the ones that can't be fixed to be ignored for now (waiting for Fyne 2.0.0 mostly). It should now be possible to set the GitHub action to be required :+1:

### CODE REVIEW

File: internal/ui/widgetpanel.go [hunk 1]
```diff
@@ -134,11 +134,8 @@ func (w *widgetPanel) showAccountMenu(from fyne.CanvasObject) {
 		w.desk.WindowManager().Close()
 	}))
 
-	popup := widget.NewPopUpMenu(fyne.NewMenu("Account", items...), root.Canvas())
-
 	bottomLeft := fyne.CurrentApp().Driver().AbsolutePositionForObject(from)
-	popup.Move(bottomLeft.Subtract(fyne.NewPos(0, popup.MinSize().Height)))
-	popup.Resize(fyne.NewSize(from.Size().Width, popup.MinSize().Height))
+	widget.ShowPopUpMenuAtPosition(fyne.NewMenu("Account", items...), root.Canvas(), bottomLeft)
```
Comment (Jacalz) [inline internal/ui/widgetpanel.go:138]: Don't know if this is the best solution either. It is what I came up with and I think it seems to work.

**Comment by Jacalz** [comment]
Staticcheck is being killed for some strange reasons. Was this what you were seeing before you turned off staticcheck? @andydotxyz

**Comment by andydotxyz** [comment]
> Was this what you were seeing before you turned off staticcheck? @andydotxyz

Yes

**Comment by Jacalz** [comment]
@dominikh Do you have any idea as to why this would be happening to staticcheck?

**Comment by dominikh** [comment]
It's being killed because of an out of memory situation. This is probably https://github.com/dominikh/go-tools/issues/825.

File: internal/ui/widgetpanel.go (hunk 1 above)
Comment (andydotxyz) [reply internal/ui/widgetpanel.go:138]: With this change is it still the same width as the widget panel?
What was the error that this change fixes?

File: screen.go [hunk 2]
```diff
@@ -29,30 +29,23 @@ type Screen struct {
 
 // CanvasScale calculates the scale for the contents of a desktop canvas on this screen
 func (s *Screen) CanvasScale() float32 {
```
[the remark this answers is not in the document]
Comment (andydotxyz) [reply screen.go:31]: I don't think so - what error is it fixing?

File: screen.go [hunk 3]
```diff
@@ -29,30 +29,26 @@ type Screen struct {
...
+			} else if scale != 0 {
+				return float32(scale)
+			}
 		}
-		fyne.LogError("Error reading scale", err)
-	}
 
-	if env != "auto" {
-		setting := fyne.CurrentApp().Settings().Scale()
-		if setting != fyne.SettingsScaleAuto && setting != 0.0 {
```
Comment (andydotxyz) [inline screen.go:53]: Like above I think we need to check against auto - that's how older versions of Fyne stored the "auto" user preferrence that is now default

**Review by andydotxyz — CHANGES_REQUESTED** [review]
I think one or two of my previous comments should have been "request changes" so applying that now

File: screen.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jacalz) [reply screen.go:31]: Sorry for a late answer. It is trying to work around the fact that `fyne.SettingsScaleAuto` is deprecated.

File: internal/ui/widgetpanel.go (hunk 1 above)
Comment (Jacalz) [reply internal/ui/widgetpanel.go:138]: Sorry for a late answer. It is trying to work around the fact that `widget.NewPopUpMenu` is deprecated. Is this just deprecated in the meaning that it no longer opens the popup on calling it? If so, then I can happily revert this :)

Comment (andydotxyz) [reply internal/ui/widgetpanel.go:138]: That is the case yes.

File: screen.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jacalz) [reply screen.go:31]: Do you have any suggestions how to solve it differently?

[the remark this answers is not in the document]
Comment (andydotxyz) [reply screen.go:31]: No. It is deprecated but if we stop using it then we lose the upgrade-path for folk who had that set in their configuration.
Either we have to copy the value (seems bad) or it needs to be marked as ignore the error

File: internal/ui/baricon.go [hunk 4]
```diff
@@ -138,7 +138,7 @@ func (bi *barIcon) TappedSecondary(*fyne.PointEvent) {
 
 	c := fyne.CurrentApp().Driver().CanvasForObject(bi)
 	pos := fyne.CurrentApp().Driver().AbsolutePositionForObject(bi)
-	menu = widget.NewPopUpMenuAtPosition(fyne.NewMenu("", []*fyne.MenuItem{addRemove}...), c, pos)
+	widget.ShowPopUpMenuAtPosition(fyne.NewMenu("", []*fyne.MenuItem{addRemove}...), c, pos)
```
Comment (andydotxyz) [inline internal/ui/baricon.go:141]: Sorry, I think this breaks the `menu.Hide` code in the `addRemove` callback. There is no reference to the menu any more

File: internal/x11/win/frame.go [hunk 5]
```diff
@@ -342,7 +342,6 @@ func (f *frame) drawDecoration(pidTop xproto.Pixmap, drawTop xproto.Gcontext, pi
 		b.SetMaximized(f.client.maximized)
 		// TODO maybe update icon?
 	}
-	f.canvas.SetScale(scale)
```
Comment (andydotxyz) [inline internal/x11/win/frame.go:345]: This line is required

**Review by andydotxyz — CHANGES_REQUESTED** [review]

**Comment by andydotxyz** [comment]
I think that if this was rebased the checks would start running again

**Comment by Jacalz** [comment]
It should now be updated against latest develop again. Going to see if I can fix these comments.

File: screen.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jacalz) [reply screen.go:31]: Marked it to be ignored for now.

File: screen.go (hunk 3 above)
Comment (Jacalz) [reply screen.go:53]: Marked as ignore for now. We can tackle this later.

File: internal/ui/baricon.go (hunk 4 above)
Comment (Jacalz) [reply internal/ui/baricon.go:141]: Good catch. It is a kind of strange way to do the code, but I have reverted this for now.

File: internal/x11/win/frame.go (hunk 5 above)
Comment (Jacalz) [reply internal/x11/win/frame.go:345]: It looks to be deprecated (`(345, 2)  SA1019  f.canvas.SetScale is deprecated: Settings are now calculated solely on the user configuration and system setup.`). Reverted and ignored for now.

**Comment by Jacalz** [comment]
Looks like we are still seeing `staticcheck` getting killed.

**Comment by andydotxyz** [comment]
So #148 would fix this maybe?

**Comment by Jacalz** [comment]
Hopefully. Lets rebase on it when that one lands :slightly_smiling_face:

**Review by andydotxyz — APPROVED** [review]

**Pull request merged**